src/ucp/proto/proto_multi.c

```diff
@@ -100,25 +98,69 @@ ucs_status_t ucp_proto_multi_init(const ucp_proto_multi_init_params_t *params,
     mpriv->lane_map   = lane_map;
     mpriv->num_lanes  = 0;
     perf.max_frag     = SIZE_MAX;
+    perf.min_length   = 0;
     ucs_for_each_bit(lane, lane_map) {
         ucs_assert(lane < UCP_MAX_LANES);
 
         lpriv     = &mpriv->lanes[mpriv->num_lanes++];
         lane_perf = &lanes_perf[lane];
-        max_frag  = ucs_double_to_sizet(lane_perf->bandwidth / max_frag_ratio,
-                                        lane_perf->max_frag);
 
         ucp_proto_common_lane_priv_init(&params->super, mpriv->reg_md_map, lane,
                                         &lpriv->super);
 
-        /* Due to floating-point accuracy, max_frag may be too large */
+        /* Calculate maximal fragment size according to lane relative bandwidth.
+           Due to floating-point accuracy, max_frag may be too large. */
+        max_frag = ucs_double_to_sizet(lane_perf->bandwidth / max_frag_ratio,
+                                       lane_perf->max_frag);
+        ucs_assert(max_frag > 0);
```

Comment:
this assert is always true, I guess you assumed that `max_frag` shoudl be `ssize_t`?

Reply:
is this protection from  == 0 or type overflow?

Reply:
yes, from ==0